core/commands/storage.go

```diff
@@ -681,13 +718,25 @@ the chunk and replies back to client for the next challenge step.`,
 		if !cfg.Experimental.StorageHostEnabled {
 			return fmt.Errorf("storage host api not enabled")
 		}
+		api, err := cmdenv.GetApi(env, req)
+		if err != nil {
+			return err
+		}
 
 		ssID := req.Arguments[0]
 		fileHash, err := cidlib.Parse(req.Arguments[1])
 		if err != nil {
 			return err
 		}
-		chunkHash := req.Arguments[2]
+		shardHash := req.Arguments[2]
+		shardCid, err := cidlib.Parse(shardHash)
+		if err != nil {
+			return err
+		}
+		shardSize, err := getContractSizeFromCid(req.Context, shardCid, api)
```

Comment:
This is (probably) not the right place to resolve a node. Host does not know if it wants to sign a contract with renter yet, but `getContractSizeFromCid` will resolve node (including downloading from renter to retrieve this information). The size should be passed in as an argument from renter, and host will verify the size during the shard get phase (after verifying other prerequisites).